ember/app/models/address.js

```diff
@@ -18,7 +18,27 @@ export default class AddressModel extends LocalizedModel {
   @attr default;
 
   get label() {
-    return [this.streetAndNumber, this.town].join(", ");
+    let address = `${this.streetAndNumber}\n`;
+
+    [
+      this.addressAddition1,
+      this.addressAddition2,
+      this.addressAddition3,
+    ].forEach((addition) => {
+      if (addition) {
+        address += `${addition}\n`;
+      }
+    });
+
+    if (this.poBox) {
+      address += `${this.intl.t(
+        "components.identity-addresses.form.label.poBox"
+      )} ${this.poBox}\n`;
+    }
+
+    address += `${this.postcode} ${this.town}\n${this.countryName}`;
+
+    return address;
```

Comment:
Nitpick: Might also work but feel free to resolve if you like your version better:
```suggestion
  return [
    this.streetAndNumber,
    this.addressAddition1,
    this.addressAddition2,
    this.addressAddition3,
    this.poBox ? `${this.intl.t("components.identity-addresses.form.label.poBox")} ${this.poBox}` : "",
    `${this.postcode} ${this.town}\n${this.countryName}`
  ].filter(Boolean).join("\n")
```

Reply:
Looks good, applied it